lib/core.js

```diff
@@ -360,22 +370,29 @@ Core.prototype.send = function(method, path, body) {
  * @returns {Promise<CoreResponse>}
  */
 Core.prototype._sendXHR = function(method, url, body) {
-  return new this.Promise(function(resolve) {
+  return new this.Promise(function(resolve, reject) {
     var r = new XMLHttpRequest();
+
     r.onreadystatechange = function() {
       if (r.readyState == 4) {
         resolve({status: r.status, text: r.responseText});
       }
     };
-
-    // Send the request.
-    r.open(method, url);
-    r.send(body);
+    try {
+      r.open(method, url);
+      r.send(body);
+    } catch(e) {
+      if (e instanceof DOMException && e.name === 'InvalidStateError') {
```

Comment:
I might not be understanding correctly, but from my testing this change and `v1` produce the same result. In both cases the `InvalidStateError` causes an attempt to call the `onRejected` callback. In `v1` this is missing so nothing happens, and in this branch it calls `reject(r)` but I'm not sure any caller handles this since `send()` is used in a "fire and forget" way, I think? Neither retries to send the request.

`r.open` followed by `r.send` seems perfectly correct so not sure why we would be getting them.